File: euler2.cpp
#include<iostream>
#include<fstream>
#include<cmath>
using namespace std;
ofstream fout("euler2.dat");
double N=500, h=0.1;
double f(double t, double y){
return y-0.5*exp(0.5*t)*sin(5*t)+5*exp(0.5*t)*cos(5*t); }
void euler1(double h, double t0, double y0){
double y=y0;
for (double t=t0;t<=5.0; t+=h){
fout<<t<<"  "<<y<<"  "<<(exp(0.5*t)*sin(5*t))<<endl;
y+=h*f(t,y); }
}
int main(){
euler1(0.1,0.0,0.0);
return 0;
}

File: euler2.py
import math

def f(t, y):
    return y - 0.5 * math.exp(0.5 * t) * math.sin(5 * t) + 5 * math.exp(0.5 * t) * math.cos(5 * t)

def euler1(h, t0, y0):
    y = y0
    t = t0
    
    with open("euler2.dat", "w") as fout:
        while t <= 5.0:
            exact = math.exp(0.5 * t) * math.sin(5 * t)
            
            # Print to console and file
            print(f"{t:.4f}  {y:.6f}  {exact:.6f}")
            fout.write(f"{t}  {y}  {exact}\n")
            
            # Euler update step
            y += h * f(t, y)
            t += h

if __name__ == "__main__":
    euler1(0.1, 0.0, 0.0)
